packages/components/src/disabled/index.js

```diff
@@ -52,30 +53,56 @@ class Disabled extends Component {
 			attributes: true,
 			subtree: true,
 		} );
+
+		document.addEventListener( 'click', this.catchEvents );
+		document.addEventListener( 'focus', this.catchEvents, true );
 	}
 
 	componentWillUnmount() {
 		this.observer.disconnect();
 		this.debouncedDisable.cancel();
+		document.removeEventListener( 'click', this.catchEvents );
+		document.removeEventListener( 'focus', this.catchEvents );
+	}
+
+	catchEvents( e ) {
+		if ( this.node && this.node.contains( e.target ) ) {
+			e.preventDefault();
+		}
 	}
 
 	bindNode( node ) {
 		this.node = node;
 	}
 
 	disable() {
```

Comment:
Looks like `this.disable` is called repeatedly via the `MutationObserver` because each time we modify a node it triggers a new mutation thereby re-triggering `this.disable`. It is going crazy!

I tried cancelling and requeueing the observer at the start of `this.disable` and then restarting it at the end but that didn't seem to do anything unless I added a timeout of 2000ms before re-adding the observer.

What I need is a local state variable to block running `this.disable` if it is already running.

Reply:
> What I need is a local state variable to block running this.disable if it is already running.

Tried this and it didn't work.

Reply:
> Looks like this.disable is called repeatedly via the MutationObserver because each time we modify a node it triggers a new mutation thereby re-triggering this.disable. It is going crazy!

Watching the code it looks like this problem is unexpected as it was handled by the usage of debounce:
```
// Debounce re-disable since disabling process itself will incur
// additional mutations which should be ignored.
```
Does adding a timeout to debounce fixes the problem?

If not, I guess we can try something safer but a little bit complex. The code that applies mutations is part of the mutation handler event what if on the mutation handler event we disable the mutation observer. We schedule the execution of the disable() operation and the disable operation after apply the mutations enables the mutation observer.
If the last suggestion also does not work I guess we can try a three scheduling approach: the mutation handler schedules a function "A". Function "A" disables the mutation observer and schedules function "B". Function "B" applies the mutations to the dom and schedules function "C". Function "C" reenables the mutation observer.

The suggestions seem hacky but if one of them works we may then understand better what's happening and may be able to find a better solution.

Reply:
I've tried some of these previously, but let me come at it again with a clear head and see how I get on. Thanks @jorgefilipecosta 👍

Reply:
Going to be looking into this soon.